### PR TITLE
fix(AutoSuggestBox): Fix Suggestion Item not being selected

### LINKED ISSUE #9576: [WASM] AutoSuggestBox still not working
### Current behavior

@jeromelaban @iury-kc AutoSuggestBox still is not working on WASM. Please see attached repo, there are 3 issues:

1) 
- start typing 'a'
- drop down comes up -> good. Now try to press CURSOR DOWN -> does not work -> bad

2) 
- click with mouse on any option in the drop down -> e.ChosenSuggestion is of type 'String', but should be 'Author' -> bad

3) (this can't be reproduced by the repo, but I see it in my app. I was unable to isolate by a repo. There must be some side effect involved, which I was unable to put my finger on)
- after setting 
```
	s.Text		= e.ChosenSuggestion.ToString();
	s.ItemsSource	= null;
```
... in OnQuerySubmitted then .OnTextChanged is fired again with e.Reason = .UserInput. This is wrong.

Honestly, this is starting to get frustrating. I was hoping after seeing [this ](https://github.com/unoplatform/uno/issues/7778)and [this ](https://github.com/unoplatform/uno/issues/7957)being fixed that we would be good now anf AutoSuggestBox would be working. Unfortunately that's not the case and the missing AutoSuggestBox support is a road block for my project.

Is there any chance to get this addressed any time soon?
[AutoSuggestBox4.zip](https://github.com/unoplatform/uno/files/9409592/AutoSuggestBox4.zip)

### Expected behavior

_No response_

### How to reproduce it (as minimally and precisely as possible)

_No response_

### Workaround

_No response_

### Works on UWP/WinUI

_No response_

### Environment

_No response_

### NuGet package version(s)

_No response_

### Affected platforms

_No response_

### IDE

_No response_

### IDE version

_No response_

### Relevant plugins

_No response_

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by TopProgrammer77**
GitHub Issue (If applicable): closes #9576

## PR Type

What kind of change does this PR introduce?

- Bugfix
## What is the current behavior?

https://github.com/unoplatform/uno/issues/9576

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [x] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/Controls/ListViewBase/ListViewBase.cs [hunk 1]
```diff
@@ -699,6 +699,7 @@ private void CleanUpAllContainers()
 			{
 				CleanUpContainer(container);
 			}
+			ItemsPanelRoot?.Children?.Clear();
```
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/Controls/ListViewBase/ListViewBase.cs:702]: Does this fix a separate bug for list view?

Comment (TopProgrammer77) [reply src/Uno.UI/UI/Xaml/Controls/ListViewBase/ListViewBase.cs:702]: **This is my behavior.**
1. Go to https://aka.platform.uno/wasm-samples-app
2. Search 'AutoSuggestBox' (search button is on the left bottom)
3. Select 'AutoSuggestBox_Keyboard'
4. Delete '0' of 'A0' in TextBox ( A0 -> A)
5. Add '0' ( A -> A0)
6. Press Down arror (  selecting should work )
7. Delete '0' and Add '0' again and Press Down arror ( selecting shouldn't work)

**Reason**
When ItemsSource are updated( i.e. containers are refreshed), some unnecessary containers are created.

the type of these unnecessary containers is object and they have not meaning.

File: src/Uno.UI/UI/Xaml/Controls/AutoSuggestBox/AutoSuggestBox.cs [hunk 2]
```diff
@@ -92,14 +94,27 @@ private void OnItemsChanged(IObservableVector<object> sender, IVectorChangedEven
...
+		{
+			if (_ItemsSource?.Any() ?? false)
+			{
+				return _ItemsSource;
+			}
+			if (Items?.Any() ?? false)
+			{
+				return Items;
+			}
+			return Enumerable.Empty<object>();
```
[the remark this answers is not in the document]
Comment (TopProgrammer77) [reply src/Uno.UI/UI/Xaml/Controls/AutoSuggestBox/AutoSuggestBox.cs:107]: So I cloned ItemsSource to _ItemsSource.

**Comment by TopProgrammer77** [comment]
https://github.com/unoplatform/uno/pull/12298/files#annotation_11250997459
I think this error seems to occur by [GetTemplate](https://github.com/unoplatform/uno/pull/12298/files#annotation_11249390264).
I tested When_Selecting_Suggest_With_UpDown_Key() in WASM on my PC and it passed.
@Youssef1313 , could you help me it?

**Comment by Youssef1313** [comment]
For now you can wrap the test in `#if !WINDOWS_UWP` I think

**Comment by TopProgrammer77** [comment]
Hello, @Youssef1313. Do the current test failures come from me?

**Comment by Youssef1313** [comment]
Likely not. I restarted the failing jobs.

File: src/Uno.UI/UI/Xaml/Controls/AutoSuggestBox/AutoSuggestBox.cs [hunk 3]
```diff
@@ -92,15 +94,28 @@ private void OnItemsChanged(IObservableVector<object> sender, IVectorChangedEven
...
 		{
 			// Calling this method before base.OnItemsSourceChanged() ensures that, in the case of an ObservableCollection, the list
 			// subscribes to CollectionChanged before AutoSuggestBox does. This is important for Android because the list needs to
 			// notify RecyclerView of collection changes before UpdateSuggestionList() measures it, otherwise we get errors like
 			// "Inconsistency detected. Invalid view holder adapter position"
-			UpdateSuggestionList();
-
 			base.OnItemsSourceChanged(e);
+			_ItemsSource = (UnwrapItemsSource() as IEnumerable)?.ToObjectArray().AsEnumerable();
+			UpdateSuggestionList();
```
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/Controls/AutoSuggestBox/AutoSuggestBox.cs:118]: This is somewhat concerning. The comment above mentions it was intentional that it's called before base.

File: src/Uno.UI/UI/Xaml/Controls/AutoSuggestBox/AutoSuggestBox.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (TopProgrammer77) [reply src/Uno.UI/UI/Xaml/Controls/AutoSuggestBox/AutoSuggestBox.cs:107]:
```
AutoSuggestBox SUT = new AutoSuggestBox();
			string[] suggestions = { "a1", "a2", "b1", "b2" };
			SUT.TextChanged += (s, e) =>
			{
				if (e.Reason == AutoSuggestionBoxTextChangeReason.UserInput)
				{
					s.ItemsSource = suggestions.Where(i => i.StartsWith(s.Text));
				}
			};
```

1. Type 'a' in TextBox of AutoSuggestBox ( TextBox.Text == a, AutoSuggestBox.ItmesSource = **{ "a1", "a2" }**, ListView: open)
2. Select "a1" on ListView with keyboard - down arrow ( TextBox.Text == a1, AutoSuggestBox.ItmesSource = **{ "a1" }**, ListView:open)

### Result:
From `s.ItemsSource = suggstions.Where(i => i.StartsWith(s.Text));`, ItemsSource changes when TextBox.Text changes, even though this change is not UserInput.
As a result, Some private collection values are changed.

File: src/Uno.UI/UI/Xaml/Controls/AutoSuggestBox/AutoSuggestBox.cs (hunk 3 above)
Comment (TopProgrammer77) [reply src/Uno.UI/UI/Xaml/Controls/AutoSuggestBox/AutoSuggestBox.cs]: This is unnecessary.

File: src/Uno.UI/UI/Xaml/Controls/AutoSuggestBox/AutoSuggestBox.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (TopProgrammer77) [reply src/Uno.UI/UI/Xaml/Controls/AutoSuggestBox/AutoSuggestBox.cs]: This change is unnecessary. This was related with https://github.com/unoplatform/uno/pull/10281

File: src/Uno.UI/UI/Xaml/Controls/ListViewBase/ListViewBase.cs (hunk 1 above)
Comment (TopProgrammer77) [reply src/Uno.UI/UI/Xaml/Controls/ListViewBase/ListViewBase.cs:702]: @Xiaoy312 , Could you review this change? Thank you.

Comment (ajpinedam) [reply src/Uno.UI/UI/Xaml/Controls/ListViewBase/ListViewBase.cs:702]: @TopProgrammer77  Thanks for the fix.

Should I be able to reproduce the issue here https://aka.platform.uno/wasm-samples-app? I am trying to reproduce it following the steps you provided but I am able to select.

I am using Firefox Developer Edition (114.0b3 (64-bit)) for my test

Comment (TopProgrammer77) [reply src/Uno.UI/UI/Xaml/Controls/ListViewBase/ListViewBase.cs:702]: Thanks @ajpinedam , If you can't reproduce on your side. you can check it on remote browser. https://live.browserstack.com/
I tested it on google 113.

Comment (TopProgrammer77) [reply src/Uno.UI/UI/Xaml/Controls/ListViewBase/ListViewBase.cs:702]: Hello, @ajpinedam . Did you reproduce the issue?

Comment (ajpinedam) [reply src/Uno.UI/UI/Xaml/Controls/ListViewBase/ListViewBase.cs:702]: @TopProgrammer77 Yes, I installed Chrome and was able to reproduce it.

Comment (ajpinedam) [reply src/Uno.UI/UI/Xaml/Controls/ListViewBase/ListViewBase.cs:702]: ✅ @TopProgrammer77  Sure!

**Comment by TopProgrammer77** [comment]
Thanks, @Xiaoy312 , Could you merge this PR also?

**Pull request merged**